manager/local.go

```diff
@@ -69,33 +73,90 @@ func (m *Manager) prepareGitRepoPath(catalog model.Catalog, update bool, catalog
 	repoPath := path.Join(m.cacheRoot, catalog.EnvironmentId, repoBranchHash)
 
 	if err := os.MkdirAll(repoPath, 0755); err != nil {
-		return "", "", catalogType, err
+		return "", "", catalogType, errors.Wrap(err, "mkdir failed")
 	}
 
 	empty, err := dirEmpty(repoPath)
 	if err != nil {
-		return "", "", catalogType, err
+		return "", "", catalogType, errors.Wrap(err, "Empty directory check failed")
 	}
 
 	if empty {
 		if err = git.Clone(repoPath, catalog.URL, branch); err != nil {
-			return "", "", catalogType, err
+			return "", "", catalogType, errors.Wrap(err, "Clone failed")
 		}
 	} else {
 		if update {
-			if git.RemoteShaChanged(catalog.URL, catalog.Branch, catalog.Commit, m.uuid) {
+			changed, err := m.remoteShaChanged(catalog.URL, catalog.Branch, catalog.Commit, m.uuid)
+			if err != nil {
+				return "", "", catalogType, errors.Wrap(err, "Remote commit check failed")
+			}
+			if changed {
 				if err = git.Update(repoPath, branch); err != nil {
-					// Ignore error unless running in strict mode
-					if m.strict {
```

Comment:
No, strict mode is still only turned on for validating catalogs. But as far as I can tell, its primary purpose is to validate the yaml files themselves. That it still controlled by the strict flag.

I removed the strict check here to accomplish the following:
If an error is encountered when updating an existing catalog that prevents us from updating the local git repo do not perform a DB update of the catalogs/templates. Previous behavior was to go ahead an update the db entries again, even though the git fetch failed.